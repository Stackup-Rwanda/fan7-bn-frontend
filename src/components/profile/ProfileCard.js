import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Avatar from './ProfileAvatar';
import Menu from './ProfileCardMenu';
import './styles/profileCard.scss';
import defaultImg from '../../assets/icons/icons8-customer-48.png';

const ProfileCard = ({
  fullName, userName, profileImg, handleNavigation,
}) => (
  <div className="profile_card">
    <Avatar src={profileImg || defaultImg} />
    <div className="profile_card__text">
      <h1 className="profile_card__text___fullname">{fullName}</h1>
      <p className="profile_card__text___username">
        @
        {userName}
      </p>
    </div>
    <Link className="profile_card__settings" to="/edit-profile">
      <p className="profile_card__settings___edit">Edit Profile</p>
      <span className="profile_card__settings___icon" />
    </Link>
    <div className="profile_card__menu">
      <Menu handleNavigation={handleNavigation} />
    </div>
  </div>
);

ProfileCard.propTypes = {
  fullName: PropTypes.string.isRequired,
  userName: PropTypes.string.isRequired,
  profileImg: PropTypes.string.isRequired,
  handleNavigation: PropTypes.func.isRequired,
};

export default ProfileCard;
