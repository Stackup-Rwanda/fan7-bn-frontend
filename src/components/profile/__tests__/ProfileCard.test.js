/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import ProfileCard from '../ProfileCard';
import Avatar from '../ProfileAvatar';
import Menu from '../ProfileCardMenu';

const history = {
  push: jest.fn(),
  replace: jest.fn(),
  createHref: jest.fn(),
};

describe('ProfileCard Component Test', () => {
  it('should return true if component exists', () => {
    const wrapper = mount(
      <Router history={history}>
        <ProfileCard />
      </Router>,
    );

    expect(wrapper.find(Avatar).exists()).toBe(true);
    expect(wrapper.find(Menu).exists()).toBe(true);
  });

  it('should render appropriate props', () => {
    const props = {
      fullName: 'Test Test',
      userName: 'testT',
    };
    const wrapper = shallow(<ProfileCard {...props} />);

    expect(wrapper.find('.profile_card__text___fullname').text()).toEqual('Test Test');
    expect(wrapper.find('.profile_card__text___username').text()).toEqual('@testT');
  });
});

describe('<ProfileCard /> shallow rendering tests', () => {
  it('matches the snapshot', () => {
    const props = {
      fullName: 'Test Test',
      userName: 'testT',
      profileImg: '',
      handleNavigation: jest.fn(),
    };
    const tree = shallow(<ProfileCard {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
