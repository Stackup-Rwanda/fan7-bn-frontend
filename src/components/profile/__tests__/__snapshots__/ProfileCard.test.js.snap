// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileCard /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="profile_card"
>
  <ProfileAvatar
    src={
      Object {
        "process": [Function],
      }
    }
  />
  <div
    className="profile_card__text"
  >
    <h1
      className="profile_card__text___fullname"
    >
      Test Test
    </h1>
    <p
      className="profile_card__text___username"
    >
      @
      testT
    </p>
  </div>
  <Link
    className="profile_card__settings"
    to="/edit-profile"
  >
    <p
      className="profile_card__settings___edit"
    >
      Edit Profile
    </p>
    <span
      className="profile_card__settings___icon"
    />
  </Link>
  <div
    className="profile_card__menu"
  >
    <ProfileCardMenu
      handleNavigation={[MockFunction]}
    />
  </div>
</div>
`;
